src/tensor.cpp

```diff
@@ -88,28 +88,20 @@ std::optional<miopenTensorLayout_t> GetDefaultLayout(unsigned num_dims)
 }
 
 template <class T>
-bool CheckLengths(const std::vector<T>& lens, T maxval = 0)
+bool CheckLengthsValues(const std::vector<T>& lens)
 {
-    if(lens.empty())
-        return false;
-    if(!std::all_of(lens.cbegin(), lens.cend(), [](T x) { return x > 0; }))
-        return false;
-    if(maxval)
-    {
-        if(!std::all_of(lens.cbegin(), lens.cend(), [maxval](T x) { return x <= maxval; }))
-            return false;
-    }
-    return true;
+    return std::all_of(lens.cbegin(), lens.cend(), [](T x) {
+        return x > 0 &&
+               static_cast<size_t>(x) <= static_cast<size_t>(std::numeric_limits<int64_t>::max());
```

Comment:
Is the compiler "smart" enough to throw away the second part of the condition for the int data type?

Reply:
Sure. Except for -O0 case.